### PR TITLE
Course membership

### LINKED ISSUE #1: Course membership
## Parent
course_service #12

## Description
To identify which courses a user has access to, a service has to be created to handle course membership.
This service should be able to add and change membership of courses for users.
Also, an end point should be exposed to allow retrieving of all courses, a user is a member of. 
## Acceptance Criteria
- [ ] compilable user service 
- [ ] existing db
- [ ] GraphQL endpoint to retrieve a list of course IDs for a user

### PR DESCRIPTION
**Pull request by BlueWaves2**
## Description of changes
Implements a representation of course membership of users.
Memberships can be created, updated and deleted.
Also, a query exists to retrieve all course memberships of a user

Closes #1 
  
## How has this been tested?
- [x] automated unit test
- [x] automated integration test
- [ ] automated acceptance test
- [ ] manual, exploratory test
    
## Checklist before requesting a review
- [x] My code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My code fulfills all acceptance criteria
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] The test coverage (line and branch) is reasonably high, especially on Service classes and other classes with complex logic
- [x] I have made corresponding changes to the documentation
- [x] I have added explanation of architectural decision and rationales to [wiki/adr](https://github.com/IT-REX-Platform/wiki/tree/main/adr)
- [x] I have updated the changes in the ticket description

## Checklist for reviewer
- [ ] The code works and does not throw errors
- [ ] The code is easy to understand and there are no confusing parts
- [ ] The code follows the [coding guidelines](https://github.com/IT-REX-Platform/wiki/blob/main/dev-manuals/backend/coding-guidelines.md) of this project
- [ ] The code change accomplishes what it is supposed to do
- [ ] I cannot think of any use case in which the code does not behave as intended
- [ ] The added and existing tests reasonably cover the code change
- [ ] I cannot think of any test cases, input or edge cases that should be tested in addition
- [ ] Description of the change is included in the documentation

### CODE REVIEW

File: src/main/resources/application-dev.properties [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Database
+spring.jpa.properties.hibernate.dialect=org.hibernate.dialect.PostgreSQLDialect
+#TODO set port and name for service database
+spring.datasource.url=jdbc:postgresql://localhost:1032/templatedatabase
```
Comment (ZombieAlienRobot) [inline src/main/resources/application-dev.properties:4]:
```suggestion
spring.datasource.url=jdbc:postgresql://localhost:5032/userdatabase
```

File: src/main/resources/application-prod.properties [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# Database
+spring.jpa.properties.hibernate.dialect=org.hibernate.dialect.PostgreSQLDialect
+#TODO set port for service database
+spring.datasource.url=jdbc:postgresql://localhost:1032/${database_name}
```
Comment (ZombieAlienRobot) [inline src/main/resources/application-prod.properties:4]:
```suggestion
spring.datasource.url=jdbc:postgresql://localhost:5032/${database_name}
```

**Review by ZombieAlienRobot — CHANGES_REQUESTED** [review]

**Review by ZombieAlienRobot — APPROVED** [review]
lgtm
